### DISCUSSION TITLE
Possible to add another drain to an existing root logger?

### DISCUSSION THREAD

**Discussion by smahm006** [slog-rs/slog#307] [Q&A]
Apologies if this is a basic question but I did not find something in the docs for this.

Let's say you built the root logger in the beginning of your program and then you needed to add a new file drain to the root logger if a certain condition is met in the middle of your program, would that be doable?

**Comment by dpc** [comment]
You can either use the existing `Logger` as a `Drain` for a new `Logger` that will replace the old one (because of https://docs.rs/slog/latest/slog/trait.Drain.html#impl-Drain-2), or you can implement your own `struct MyMagicDrain` that `impl Drain for MyMagicDrain` and inside do some magic to conditionally log to different places. There's also https://docs.rs/slog-atomic/

**Reply by smahm006** [reply]
Hey thanks a lot for the reply. I will definitely look into your suggestions, I really like the idea of using the existing logger as a drain for the new logger so most likely I will go that route. I appreciate the response again and thank you for such a great crate.
